specs/dwritecore/DWriteCore.md

```diff
@@ -0,0 +1,255 @@
+
+# Description
+
+New APIs introduced by DWriteCore include:
+
+ - A new DWRITE_FACTORY_TYPE enumeration for creating a _restricted factory_ object
+ - A new method for getting pixel data from a bitmap render target
+
+The new DWRITE_FACTORY_TYPE_RESTRICTED enum value complements the existing values for creating a _shared factory_
+and _isolated factory_ objects. A restricted factory is more locked-down than either of the existing factory
+types. First, it only caches data in-process, and will not attempt to either read from or write to a cross-process
+font cache or persisted cache file. Second, a restricted factory only enumerates well-known fonts in the system
+font collection. The well-known-fonts restriction enables DWriteCore to efficiently construct the system font 
+collection without depending on a cross-process or persistent cache. An example use case for a restricted factory
+would be a web browser rendering process with very limited permissions.
+
+A _bitmap render target_ (IDWriteBitmapRenderTarget) is an API object that encapsulates a system memory bitmap
+and enables rendering glyphs to the bitmap. On Windows, the object encapsulates a GDI memory DC with a GDI
+device-dependent bitmap (DIB) selected into it. However, on non-Windows platforms, there is no GDI DC,
+and the bitmap is just a system memory array. DWriteCore introduces a new version of the bitmap render target
+interface that an application can use to access the bitmap data without going through GDI.
+
+# Examples
+
+## Creating a restricted factory
+
+An application calls the [DWriteCreateFactory](https://docs.microsoft.com/en-us/windows/win32/api/dwrite/nf-dwrite-dwritecreatefactory)
+function to create a factory object, which is the entry-point to all other DirectWrite APIs. An application can
+create a _restricted factory_ if it does not want the factory or objects created from it to read from or write
+to any cross-process or persistent cache. In addition, a restricted factory's system font collection only includes
+well-known system fonts.
+
+The following example creates a restricted factory.
+
+```c++
+ComPtr<IDWriteFactory7> spFactory;
+HRESULT hr = DWriteCreateFactory(
+    DWRITE_FACTORY_TYPE_RESTRICTED,
+    __uuidof(IDWriteFactory7),
+    (IUnknown**)&spFactory
+    );
+```
+
+## Drawing glyphs to a system memory bitmap
+
+An application creates a bitmap render target by calling [IDWriteGdiInterop::CreateBitmapRenderTarget](https://docs.microsoft.com/en-us/windows/win32/api/dwrite/nf-dwrite-idwritegdiinterop-createbitmaprendertarget).
+A bitmap render target encapsulates a bitmap in system memory, and eanbles rendering glyphs to the bitmap. On Windows,
+the bitmap is actually a GDI DIB selected into a memory HDC, and it is possible to get the bitmap pixels using GDI
+functions. However, DWriteCore introduces a simpler method through the new IDWriteBitmapRenderTarget2 interface.
+The following example shows both the old way and the new way.
```

Comment:
The structure name `DWRITE_BITMAP_DATA_BGRA32` tells me that the pixel format is BGRA32. But what about the other attributes? From reading the documentation for `IDWriteBitmapRenderTarget::GetDC`, I infer that the underlying bitmap data returned by `DWriteBitmapRenderTarget2::GetBitmapData` is

1. A top-down bitmap,
2. with no padding between scan lines.

Should be documented more clearly what format to expect.

Reply:
DISCUSSION: It's top-down, 32bpp without scanline padding. (There's a bug in the sample code.)  What if someone wants another format in the future?  We'd create a new API, new input/output to handle it.  For non-breaking changes it could be a `...2` method, for breaking-change boundaries it could be the same method but re-typed so consumers detect the change as a compilation error.

RECOMMEND: Contact the target consumer groups and ask them for feedback.

Reply:
SAMPLE: Can we stuff a composition render target into DWrite to target?

Reply:
I added clarification that it's top-down with no scanline padding.

Regarding composition render target, we're still discussing how to support more powerful rendering (like Direct2D offers).